src/main/java/io/excaliburfrc/robot/commands/AutomaticIntakeCommand.java

```diff
@@ -0,0 +1,33 @@
+package io.excaliburfrc.robot.commands;
+
+import edu.wpi.first.math.trajectory.Trajectory;
+import edu.wpi.first.math.trajectory.TrajectoryGenerator;
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import io.excaliburfrc.robot.subsystems.Drive;
+import io.excaliburfrc.robot.subsystems.Intake;
+import io.excaliburfrc.robot.subsystems.Vision;
+import java.util.Arrays;
+
+public class AutomaticIntakeCommand extends CommandBase {
+  private final Drive drive;
+  private final Intake intake;
+  private final Vision vision;
+  private Trajectory trajectory;
+
+  public AutomaticIntakeCommand(Drive drive, Intake intake, Vision vision) {
+    this.drive = drive;
+    this.intake = intake;
+    this.vision = vision;
+  }
+
+  @Override
+  public void initialize() {
+    raceWith(intake.intakeBallCommand());
+    var currentPose = drive.getPose();
+    trajectory =
+        TrajectoryGenerator.generateTrajectory(
+            Arrays.asList(currentPose, currentPose.plus(vision.getTransform())),
+            drive.trajectoryConfig);
+    andThen(drive.followTrajectoryCommand(trajectory));
+  }
```

Comment:
Whatever you're doing here, it definitely shouldn't be in `initialize()`. When extending command classes, you should (nearly) *never* override the lifecycle methods.

`SelectCommand` might be relevant for this.